### ISSUE TITLE
[no-redeclare] ignoreDeclarationMerge when using enum+namespace

### ISSUE THREAD

**Issue by idan-at** [typescript-eslint/typescript-eslint#3568]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```ts
enum E {
  A = "a",
  B = "b"
}

namespace E {
  export function values(): string[] {
    return [E.A, E.B];
  }
}
```

```JSON
{
  "rules": {
     "@typescript-eslint/no-redeclare": ["error", { "ignoreDeclarationMerge": true }],
  }
}
```

**Expected Result**

I'd expect the rule not to emit an error in this case.

**Actual Result**

```
 6:11  error  'E' is already defined                                                                @typescript-eslint/no-redeclare
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.27.0` |
| `@typescript-eslint/parser`        | `4.27.0` |
| `TypeScript`                       | `4.3.0` |
| `ESLint`                           | `7.29.0` |
| `node`                             | `14.16.1` |

**Comment by bradzacher** [comment]
I'm curious - why are you declaration merging such a function into your enum when you can just leverage `Object.values`?

```
enum E {
  A = "a",
  B = "b"
}

const x = Object.values(E); // typeof x === E[]
```

**Comment by idan-at** [comment]
Simply for values you’re right, but I can think of other use cases that fit. For example, an ‘exists’ that makes sure your value is a valid enum variant.

I probably wouldn’t write that code from scratch, but when migrating existing JavaScript code to TS, this pattern can be useful :)
